pandas/core/groupby/ops.py

```diff
@@ -914,20 +916,26 @@ def agg_series(
         np.ndarray or ExtensionArray
         """
 
-        if not isinstance(obj._values, np.ndarray):
+        result = self._aggregate_series_pure_python(obj, func)
+
+        if isinstance(obj._values, ArrowExtensionArray):
+            return maybe_cast_to_pyarrow_result(result, obj.dtype)
```

Comment:
Can you use `maybe_cast_pointwise_result` with `same_dtype=False` here instead.

Reply:
resolved